### PR TITLE
Add @sunxiaoguang to MAINTAINERS.md

### PR DESCRIPTION
**Pull request by sunxiaoguang**
My name is Sun Xiaoguang from Zhihu, where I help make a BigTable like system focusd on OLTP that is using TiKV. I've been contributing to TiKV and the library ecosystem for around a year now. I first touched the TiKV codebase since around March 2018.

As part of my involvement with the project I've contributed several improvements to the TiKV Project:

* [Added raw batch put/get/delete/scan](https://github.com/pingcap/kvproto/pull/244)
* [Changed `get_region` and `get_region_info` to use a common implementation](https://github.com/tikv/tikv/pull/3521)
* [Print help message when starting tikv-ctl without argument](https://github.com/tikv/tikv/pull/3531)
* [Clean up unused #[allow] directives](https://github.com/tikv/tikv/pull/3553)

I also am the first and primary author of the [TiKV Rust Client](https://github.com/tikv/client-rust). You can see related PRs for that here:

* [TiKV Rust Client RFC](https://github.com/tikv/rfcs/pull/7)
* [The initial version of Raw KV Implementation](https://github.com/tikv/client-rust/pull/14)
* [Change `raw::Client::get` to return `Option<Value>`](https://github.com/tikv/client-rust/pull/24)
* [Remove cf and batch_scan from `example/raw.rs`](https://github.com/tikv/client-rust/pull/21)

I also made the first PR to the mock TiKV project: [A basic implementation of mock TiKV API surface](https://github.com/tikv/mock-tikv/pull/1)

As part of my involvement, the TiKV team inquired if I was interested in becoming a maintainer for the project.

This PR adds me to the list of maintainers. I have read and understand the expectations of maintainers described in the `GOVERNANCE.md` file.

Signed-off-by: Xiaoguang Sun <sunxiaoguang@zhihu.com>

### CODE REVIEW

File: MAINTAINERS.md [hunk 1]
```diff
@@ -10,3 +10,4 @@ This file lists who are the core maintainers of the TiKV project.
 
 * [Jay Li](https://github.com/busyjay); jay@pingcap.com; Focus Areas: Raft, Coprocessor
 * [Jinpeng Zhang](https://github.com/zhangjinpeng1987); zhangjinpeng@pingcap.com; Focus Areas: Storage, Transaction
+* [Xiaoguang Sun](https://github.com/sunxiaoguang); sunxiaoguang@zhihu.com; Focus Areas: Storage, Search Engine
```
Comment (breeswish) [inline MAINTAINERS.md]: Congratulations! BTW Focus area should be the role that you involve in the TiKV project.

**Comment by Hoverbear** [comment]
Hi @sunxiaoguang ! 

As we discussed in person over the last months, we'd be very happy to welcome you to the team!

We're going to update the format of this list and have it include all folks with commit rights. For the time being, this looks fine to me. For your focus areas maybe "Rust Client, Mock TiKV" are appropriate?

**Comment by siddontang** [comment]
I think the areas of Sun is "Storage, Client", the Client includes Rust client and Mock TiKV.

File: MAINTAINERS.md (hunk 1 above)
Comment (sunxiaoguang) [reply MAINTAINERS.md]: I see. Let me fix it.

**Comment by sunxiaoguang** [comment]
@siddontang @breeswish @Hoverbear 
Thanks everybody for your help in the past, and I'm very excited and looking forward to contribute more to TiKV project.

**Comment by Hoverbear** [comment]
@sunxiaoguang We updated the format, could you rebase and put your name in "Maintainers"?

File: MAINTAINERS.md [hunk 2]
```diff
@@ -20,7 +20,7 @@ Name | Email | Focus
 
 Name | Email | Focus
 ----|---|---
-
+[[Xiaoguang Sun](https://github.com/sunxiaoguang)] | [sunxiaoguang@zhihu.com](mailto:sunxiaoguang@zhihu.com) | Focus Areas: Client, Storage
```
Comment (kennytm) [inline MAINTAINERS.md]:
```suggestion
[Xiaoguang Sun](https://github.com/sunxiaoguang) | [sunxiaoguang@zhihu.com](mailto:sunxiaoguang@zhihu.com) | Focus Areas: Client, Storage
```

**Comment by BusyJay** [comment]
This PR still needs a approval from either @zhangjinpeng1987 or @siddontang right? /cc @Hoverbear

**Comment by Hoverbear** [comment]
Yes unfortunately @breeswish is not one of our senior maintainers. :)

**Pull request merged**